microcosm_pubsub/registry.py

```diff
@@ -90,8 +93,25 @@ def __getitem__(self, media_type):
         Create a codec or raise KeyError.
 
         """
-        schema_cls = self.mappings[media_type]
-        return PubSubMessageCodec(schema_cls(strict=self.strict))
+        try:
+            schema_cls = self.mappings[media_type]
+            if isinstance(schema_cls, string_types):
+                media_type = schema_cls
+                schema = URIMessageSchema(media_type)
+            else:
+                schema = schema_cls(strict=self.strict)
+        except KeyError:
+            if not self.auto_register:
+                raise
+
+            for lifecycle_change in LifecycleChange:
```

Comment:
If we miss the registry and have auto registering enabled, we can generate schema instance on the fly.